course_discovery/apps/api/serializers.py

```diff
@@ -266,7 +266,7 @@ def choices(self):
         return OrderedDict(sorted([(x.slug, x.name) for x in Subject.objects.all()], key=lambda x: x[1]))
 
     def get_number_of_courses(self, obj):
-        return obj.course_set.count()
+        return CourseRun.objects.filter(course__subjects=obj, status=CourseRunStatus.Published).count()
```

Comment:
Is it working? If yes, I'm wondering how?? 🤔
In my sight it should be like this:
```
CourseRun.objects.filter(course__subjects__in=obj, status__slug=CourseRunStatus.Published).count()
```

Reply:
@iam-mhaseeb its working fine 
![image](https://user-images.githubusercontent.com/7139602/121336194-b9588280-c934-11eb-980d-a2e71de95311.png)

where this query `CourseRun.objects.filter(course__subjects__in=obj).count()`  return error `TypeError: 'Subject' object is not iterable`

Reply:
There could be multiple subjects attached to one course, this is actually concerning me and we are just checking a single object.